doc: Update known issues for NCS 3.1.0 (#23586)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/known_issues.rst

Bluetooth LE
============

> 

Author: Added.

> .. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-2 v2-9-1 v2-9-0 v2-8-0

Reviewer: ```suggestion
.. rst-class:: v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-2 v2-9-1 v2-9-0 v2-8-0
```
This is fixed in 3.1.0.

  When the Host and Controller run on separate cores, the ``<err> bt_hci_driver: No available ACL buffers!`` error might be printed.

  **Workaround:** Increase the value of the :kconfig:option:`CONFIG_BT_BUF_ACL_RX_COUNT_EXTRA` Kconfig option.
  This workaround is only applicable for the |NCS| v3.0.2, v3.0.1 and v3.0.0 releases.

> 

Author: Added.

NCSDK-31487: Improper reuse of dynamic GATT service handles
  The Host reuses handles when registering dynamic services even while there are connected clients.
  This violates the Bluetooth Core Specification and creates a theoretical risk that a connected ATT client may send requests intended for an old service, which could be misinterpreted as requests for a new service.
  The issue occurs when unregistering and then registering services using the :c:func:`bt_gatt_service_unregister` and :c:func:`bt_gatt_service_register` functions while clients remain connected.

  **Workaround:** For a simple approach, register new services only while there are no connections or connectable advertisers.
  For a more flexible approach, you may register services while connected, but after unregistering any service, wait until all connections are gone and there are no connectable advertisers before registering new services.

>   Alternatively, manually allocate handles by setting the :c:member:`~bt_gatt_attr.handle` field in each :c:type:`bt_gatt_attr` before calling the :c:func:`bt_gatt_service_register` function to ensure handles are not reused while ATT bearers exist.

Reviewer 2: ```suggestion
  Alternatively, manually allocate handles by setting the :c:member:`bt_gatt_attr.handle` field in each :c:type:`bt_gatt_attr` before calling the :c:func:`bt_gatt_service_register` function to ensure handles are not reused while ATT bearers exist.
```

  This does not guarantee that the problem is solved, as multiple blocking calls might exhaust the buffer pool.
  You can also use the (experimental) :kconfig:option:`CONFIG_BT_CONN_TX_NOTIFY_WQ` Kconfig option to use a separate workqueue for connection TX notify processing.

.. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-2 v2-9-1 v2-9-0 v2-8-0 v2-7-0 v2-6-4 v2-6-3 v2-6-2 v2-6-1 v2-6-0

> DRGN-24352: Missing disconnection events when HCI Controller to host flow control is enabled

Reviewer 3: DRGN-24352 has been renamed to NCSDK-32999. To better track this issue, please update the ticket name

The issues in this section are related to the :ref:`ug_matter` protocol.

> .. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-2 v2-9-1 v2-9-0 v2-8-0 v2-7-0

Reviewer 4: This issue is not present in 3.1.0 anymore.

KRKNWK-19277: Invalid testing steps in the Light Switch README file
  The testing steps for groupcast binding are invalid in the :ref:`matter_light_switch_sample` sample documentation.

  **Workaround:** For groupcast binding, refer to the :ref:`ug_matter_group_communication` user guide instead of the sample documentation.

> .. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-2 v2-9-1 v2-9-0 v2-8-0 v2-7-0

Reviewer 4: This issue is not present in 3.1.0 anymore.

  **Workaround:** Remove the identify cluster from the :file:`.zap` file, as this cluster is optional for the aggregator endpoint type.

> .. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-2 v2-9-1 v2-9-0

Reviewer 4: This issue is not present in 3.1.0 anymore.

        <arg name="arg1" type="int8u"/>
      </command>

> .. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-1 v2-9-0 v2-8-0 v2-7-0

Reviewer 4: This issue is not present in 3.1.0 anymore.

       git -C modules/lib/matter submodule sync third_party/pigweed/repo
       west update

Near Field Communication (NFC)
==============================

The issues in this section are related to the :ref:`ug_nfc` protocol.

.. rst-class: v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-2 v2-9-1 v2-9-0 v2-8-0 v2-7-0

NCSDK-34404: Possible increased current consumption in NFC applications on the nRF54H20 and nRF54L devices

>   If your application uses NFC on the nRF54H20 and nRF54L SoCs, the current consumption may increase after the NFC field is lost, due to the running high-frequency peripheral clock.

Reviewer 5: ```suggestion
  If your application uses NFC on the nRF54H20 and nRF54L SoCs, the current consumption may remain high even after the NFC field is lost, due to the running high-frequency peripheral clock.
```

The issues in this section are related to the :ref:`ug_wifi` protocol.

.. rst-class:: v3-1-0

SHEL-3745: Wi-Fi connection broken with nRF54H20
  Wi-Fi connection fails due to ``pbkdf2_sha1()`` error.

  **Affected platforms:** nRF7002, nRF54H20

.. rst-class:: v3-1-0

NRF7X-323: Wi-Fi ``ns`` builds take eight seconds to boot
  It takes eight seconds when autoloading credentials before the scan/connect procedure occurs.

>   **Affected platforms:** nRF7002

Reviewer 6: this issue is applicable from 2.8.0 onwards.

.. include:: /includes/zigbee_deprecation.txt

> .. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-2 v2-9-1 v2-9-0 v2-8-0 v2-7-0 v2-6-4 v2-6-3 v2-6-2 v2-6-1 v2-6-0 v2-5-3 v2-5-2 v2-5-1 v2-5-0 v2-4-4 v2-4-3 v2-4-2 v2-4-1 v2-4-0 v2-3-0 v2-2-0 v2-1-4 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0

Reviewer 7: For the entire Zigbee section we stopped updating known issues after 3.0.0, the tags can be removed (see note in the line 1492). FYI @mariusz-nordicsemi

    nRF Desktop is also affected by :ref:`hogp_readme` library's issue :ref:`NCSDK-30288 <ncsdk_30288>`.
    The issue might cause accessing data under the ``NULL`` pointer in case of Bluetooth disconnection while forwarding a configuration channel operation in the :ref:`nrf_desktop_hid_forward`.

> 

Reviewer 8: Please add a new known issue:

```
.. rst-class:: v3-1-0

NCSDK-34299: nRF Desktop application does not build or run due to the IronSide SE migration
  The :ref:nrf_desktop application is currently incompatible with the latest |NCS| transition to the IronSide SE solution for the nrf54h20dk/nrf54h20/cpuapp board target.
  As a result, this application cannot be used with that board target in any of the affected releases.

  **Affected platforms:** nRF54H20

  **Workaround:** Use a future |NCS| release that includes IronSide SE support in the :ref:nrf_desktop application.
```

Author: Added.

       reg = <0xf6000 DT_SIZE_K(28)>;
     };

> .. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-2 v2-9-1 v2-9-0 v2-8-0 v2-7-0

Reviewer 9: Please remove the v3-1-0 entry. The issue has been fixed.

Reviewer 9: Thanks!

Bluetooth samples
=================

> .. rst-class:: v3-0-2 v3-0-1 v3-0-0

Reviewer 8: Please add a new issue:

.. rst-class:: v3-1-0

NCSDK-34821: Fast Pair samples do not build or run due to the IronSide SE migration
  The :ref:`fast_pair_input_device` and :ref:`fast_pair_locator_tag` samples are currently incompatible with the latest |NCS| transition to the IronSide SE solution - bundle 22.1.0 and higher for the ``nrf54h20dk/nrf54h20/cpuapp`` board target.
  As a result, you cannot use these samples with that board target in any of the affected releases.

  **Affected platforms:** nRF54H20

  **Workaround:** Use a future |NCS| release that includes IronSide SE support in the Fast Pair samples or use the |NCS| v3.0.0 release with the SUIT SDSC bundle (version up to 0.9.6).

  * :ref:`fast_pair_input_device`
  * Fast Pair configurations of the :ref:`nrf_desktop` application

> .. rst-class:: wontfix v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-2 v2-9-1 v2-9-0 v2-8-0 v2-7-0 v2-6-4 v2-6-3 v2-6-2 v2-6-1 v2-6-0 v2-5-3 v2-5-2 v2-5-1 v2-5-0 v2-4-3 v2-4-2 v2-4-1 v2-4-0 v2-3-0 v2-2-0 v2-1-4 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0 v1-5-2 v1-5-1 v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0 v0-4-0 v0-3-0

Reviewer 9: Please add:
```
.. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-2 v2-9-1 v2-9-0 v2-8-0 v2-7-0 v2-6-4 v2-6-3 v2-6-2 v2-6-1 v2-6-0 v2-5-3 v2-5-2 v2-5-1 v2-5-0 v2-4-3 v2-4-2 v2-4-1 v2-4-0 v2-3-0 v2-2-0 v2-1-4 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0 v1-5-2 v1-5-1 v1-5-0 v1-4-2 v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0 v0-4-0 v0-3-0

NCSDK-34682: Reconnection with HID devices that have privacy enabled might not work on Android 16
  If an Android 16 device is disconnected from the HID device, the Android device faces issues.
  The smartphone automatically reconnects with the HID device, but in the Android settings, it says that the phone is not connected, and HID reports are ignored.
  When you click on the connect button it will be stuck in the connecting loop which will last for tens of seconds or in most cases forever.
  See the following list of affected samples and applications:
  * :ref:`peripheral_hids_mouse`
  * :ref:`fast_pair_input_device`
  * Fast Pair configurations of the :ref:`nrf_desktop` application
  **Workaround:** On :ref:`peripheral_hids_mouse` you can disable the :kconfig:option:`CONFIG_BT_PRIVACY` Kconfig option.
```

Reviewer 9: Thanks!

  The NVM size of the nRF54L10 SoC is equal to 1012 KB.

  The workaround fix for the nRF54L10 partition map is a breaking change and cannot be performed using DFU.
  The DFU procedure will fail if you attempt to upgrade the sample firmware based on one of the |NCS| releases that is affected by this issue.

  **Affected platforms:** nRF54L10

  **Workaround:** Manually cherry-pick and apply the commit with the fix from the ``main`` branch of the ``sdk-nrf`` repository (commit hash: ``43d0e2eb129bde7b0f0fc462dd4b239765c69c59``).
  Additionally, you need to manually update the :file:`boards/nordic/nrf54l15dk/nrf54l15dk_nrf54l10_cpuapp.dts` DTS file in your local copy of the ``sdk-zephyr`` repository to avoid issues with build asserts that are used for RRAM validation.
  Correct the ``cpuapp_rram`` DTS node in the following way:

  .. code-block::

     &cpuapp_rram {
       reg = <0x0 DT_SIZE_K(1012)>;
     };

  Then, correct the ``storage_partition`` node in the following way:

  .. code-block::

     storage_partition: partition@f6000 {
       reg = <0xf6000 DT_SIZE_K(28)>;
     };

> .. note::

Author: Moved.

Cellular samples
================

> .. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-2 v2-9-1 v2-9-0 v2-8-0

Reviewer 10: ```suggestion
.. rst-class:: v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-2 v2-9-1 v2-9-0 v2-8-0
```

Reviewer 10: this has been marked resolved as of Jan-2025

Peripheral samples
==================

> .. rst-class:: v3-0-2 v3-0-1 v3-0-0

Reviewer 5: Please add a new issue:
```
.. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0

NCSDK-30519: The :ref:`radio_test` sample reports high packet error rate on the long-range data rates.
  Packet error rate is very high in the receive mode when using the ``BLE_LR125KBIT`` and ``BLE_LR500KBIT`` data rates.

  **Affected platforms:** nRF54L15, nRF54L10, nRF54L05
```

Author: Added.

> .. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0

Author: v3-1-0 removed

The issues in this section are related to :ref:`app_bootloaders`.

.. rst-class:: v3-1-0

NCSDK-19265: Errors and faults in retrieving the NV counter results in disabled rollback protection
  The rollback protection in NSIB does not validate counter values if the values cannot be retrieved correctly or if they are larger than the largest supported value.
  In such a case, the image booted might be one whose values is lower than expected NV counter value.

.. rst-class:: v3-1-0

> NCSDK-31918: Monotonic counter update protection does not work

Reviewer 11: ```suggestion
NCSDK-31918: NSIB active slot function can give invalid results
```

  In such a case, the image booted might be one whose values is lower than expected NV counter value.

.. rst-class:: v3-1-0

NCSDK-31918: NSIB active slot function can give invalid results
  The function for getting the active s0 or s1 slot only checks the :c:struct:`fw_info` structure.
  The image itself is not validated nor is NSIB queried for which slot is active.
  This means that if an image is invalidated but has a higher firmware version or if the image is corrupt or even entirely missing (except for the :c:struct:`fw_info` structure), the response to this query will return the wrong image, possibly leading to the wrong update being loaded or impossibility to load a firmware update (depending upon firmware configuration).

.. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-2 v2-9-1 v2-9-0 v2-8-0 v2-7-0 v2-6-4 v2-6-3 v2-6-2 v2-6-1 v2-6-0 v2-5-3 v2-5-2 v2-5-1 v2-5-0 v2-4-4 v2-4-3 v2-4-2 v2-4-1 v2-4-0 v2-3-0 v2-2-0 v2-1-4 v2-1-3 v2-1-2 v2-1-1 v2-1-0 v2-0-2 v2-0-1 v2-0-0 v1-9-2 v1-9-1 v1-9-0 v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0

> NCSDK-31918: Monotonic counter update protection counter limit does not work

Reviewer 11: ah I just noticed `NCSDK-31918` is already listed here :|, could you update description with the `NCSDK-19265` contents from above then change `NCSDK-31918` above to `NCSDK-19265` and also put the same version numbers from here to the above bug? Sorry

The issues in this section are related to :ref:`app_dfu`.

> .. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1

Reviewer 12: ```suggestion
.. rst-class:: v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1
```

SUIT removed from codebase

  **Workaround:** Enable protection manually using the :kconfig:option:`CONFIG_FPROTECT` and  :kconfig:option:`CONFIG_FPROTECT_ALLOW_COMBINED_REGIONS` Kconfig options of MCUboot.

> .. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0

Reviewer 12: Won't fix

The issues in this section are related to :ref:`MCUboot <mcuboot_wrapper>`.

.. rst-class:: v3-1-0

NCSDK-34594: MCUboot might hang and is failing on FPROTEC assertion when using external flash

>   FPROTEC can support locking up to 62 kB of RRAM on nRF54L SoCs.

Reviewer 13: maybe wort to add at the end of sentence: ... when used by immutable MCUboot.

  Either the partition declared for MCUboot or its size might be too large for using FPROTECT (depends on MCUboot configuration).

>   Without access to MCUboot, this might be misrecognized by an application.

Reviewer 13: Without access to MCUboot Logs, this might be misrecognized as not working/starting application.

The issues in this section are related to :ref:`nrfxlib:mpsl`.

> .. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0 v2-9-0-nRF54H20-1 v2-9-1 v2-9-0 v2-8-0

Reviewer 3: DRGN-25261 has been fixed and does not affect v3.1

Other issues
************

> .. rst-class:: v3-1-0 v3-0-2 v3-0-1 v3-0-0

Author: Fixed.
